src/Tools/DynamoShapeManager/Utilities.cs

```diff
@@ -147,6 +147,10 @@ public static Version GetInstalledAsmVersion2(IEnumerable<Version> versions, ref
                 getASMInstallsFunc = getASMInstallsFunc ?? GetAsmInstallations;
                 var installations = getASMInstallsFunc(rootFolder);
 
+                //filter install locations missing tsplinesA
+                installations = installations.Cast<KeyValuePair<string, Tuple<int,int,int,int>>>().
+                    Where(x => Directory.EnumerateFiles(x.Key, "tsplines*A.dll").Any());
```

Comment:
Does it make sense to fold this check into `getASMInstallsFunc`?

Reply:
1. I can try to fold it in there - 
2. we're already checking for that in the findProductInstallations function when we pass `"ASMAHL*.dll"` isn't this - essentially doing the same check twice?

Reply:
There could be an "ASMAHL*A.dll" and an "ASMAHL*I.dll"?

Reply:
kind of, the offending folder has the `ASMA.dlls` and `tsplinesI.dll `